src/jobsautoreport/slack.py

```diff
@@ -14,87 +14,116 @@ def __init__(self, web_client: WebClient, channel_id: str) -> None:
         self._client = web_client
         self._channel_id = channel_id
 
-    def send_report(self, report: Report) -> None:
-        blocks = self._format_message(report)
-        logger.debug("Message format finished successfully")
+    def _post_message(
+        self,
+        report: Report,
+        format_function: Callable[[Report], list[dict[str, Any]]],
+        thread_time_stamp: Optional[str],
+    ) -> str:
+        blocks = format_function(report)
+        logger.debug("Message formatted successfully")
         response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks
+            channel=self._channel_id, blocks=blocks, thread_ts=thread_time_stamp
         )
         response.validate()
         logger.info("Message sent successfully")
+        return response["ts"]
 
-        time_stamp = response["ts"]
-        blocks = self._format_presubmit_comment(report)
-        logger.debug("comment format finished successfully")
-        response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks, thread_ts=time_stamp
+    def _upload_file(
+        self,
+        file_title: str,
+        file_path: str,
+        filename: str,
+        thread_time_stamp: Optional[str],
+    ) -> None:
+        response = self._client.files_upload(
+            channels=[self._channel_id],
+            file=file_path,
+            filename=filename,
+            initial_comment=file_title,
+            thread_ts=thread_time_stamp,
         )
         response.validate()
-        logger.info("Presubmit comment sent successfully")
-        blocks = self._format_equinix_comment(report)
-        logger.debug("comment format finished successfully")
-        response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks, thread_ts=time_stamp
+        logger.info("%s was uploaded successfully", filename)
+
+    def send_report(self, report: Report) -> None:
+        thread_time_stamp = self._post_message(
+            report=report,
+            format_function=self._format_message,
+            thread_time_stamp=None,
+        )
+        self._create_and_upload_success_image(
+            jobs=report.top_10_failing_e2e_or_subsystem_periodic_jobs,
+            file_title="Top 10 Failed Periodic Jobs",
```

Comment:
it's counter-intuitive to call it ``_create_and_upload_success_image`` if it contains mostly failing jobs
``_create_and_upload_success_image`` -> ``_upload_most_failing_jobs_graph``